diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index 0cae20f09aba..f3cf9af03cfb 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -414,6 +414,8 @@ static int read_sr(struct spi_nor *nor)
 		return ret;
 	}
 
+        nor->bouncebuf[0] &= ~SR_WIP;  // Roy clear the WIP bit 
+        
 	return nor->bouncebuf[0];
 }
 
@@ -802,6 +804,7 @@ static int spi_nor_fsr_ready(struct spi_nor *nor)
 
 static int spi_nor_ready(struct spi_nor *nor)
 {
+        
 	int sr, fsr;
 
 	if (nor->flags & SNOR_F_READY_XSR_RDY)
@@ -813,6 +816,7 @@ static int spi_nor_ready(struct spi_nor *nor)
 	fsr = nor->flags & SNOR_F_USE_FSR ? spi_nor_fsr_ready(nor) : 1;
 	if (fsr < 0)
 		return fsr;
+	
 	return sr && fsr;
 }
 
@@ -5006,6 +5010,8 @@ static int spi_nor_probe(struct spi_mem *spimem)
 	if (!nor->mtd.name)
 		nor->mtd.name = spi_mem_get_name(spimem);
 
+        printk(KERN_ERR "Roy %s at %s %d \n", nor->mtd.name, __FUNCTION__, __LINE__);
+
 	/*
 	 * For some (historical?) reason many platforms provide two different
 	 * names in flash_platform_data: "name" and "type". Quite often name is
@@ -5020,6 +5026,7 @@ static int spi_nor_probe(struct spi_mem *spimem)
 		flash_name = spi->modalias;
 
 	ret = spi_nor_scan(nor, flash_name, &hwcaps);
+	
 	if (ret)
 		return ret;
 
diff --git a/include/linux/mtd/spi-nor.h b/include/linux/mtd/spi-nor.h
index fc0b4b19c900..887da3672d82 100644
--- a/include/linux/mtd/spi-nor.h
+++ b/include/linux/mtd/spi-nor.h
@@ -243,6 +243,7 @@ enum spi_nor_option_flags {
 	SNOR_F_4B_OPCODES	= BIT(6),
 	SNOR_F_HAS_4BAIT	= BIT(7),
 	SNOR_F_HAS_LOCK		= BIT(8),
+	SNOR_F_IGNORE_WIP       = BIT(9),
 };
 
 /**
