diff --git a/include/configs/imx8mm_var_dart.h b/include/configs/imx8mm_var_dart.h
index 5fcebcc0e6..547cf88a24 100644
--- a/include/configs/imx8mm_var_dart.h
+++ b/include/configs/imx8mm_var_dart.h
@@ -73,7 +73,7 @@
 	"fdt_high=0xffffffffffffffff\0"		\
 	"boot_fdt=try\0" \
 	"ip_dyn=yes\0" \
-	"fdt_file=undefined\0" \
+	"fdt_file=imx8mm-resptech.dtb\0" \
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
 	"mmcblk=1\0" \
 	"mmcautodetect=yes\0" \
@@ -158,6 +158,11 @@
 
 #define CONFIG_BOOTCOMMAND \
 	"run ramsize_check; " \
+	"if gpio input 123; then " \
+	  "echo GPIO Active; " \
+	"else " \
+	  "echo GPIO Inactive; " \
+        "fi; " \
 	"mmc dev ${mmcdev}; "\
 	"if mmc rescan; then " \
 		"if test ${use_m4} = yes && run loadm4bin; then " \
